.github/actions/test-report/action.yml

```diff
@@ -59,7 +60,19 @@ inputs:
 runs:
   using: "composite"
   steps:
+    - name: Report go test results
+      if: contains('go-json', inputs.reporter)
+      uses: elastic/apm-pipeline-library/.github/actions/test-go-report@feature/go-json-reporter
+      with:
+        name: ${{ inputs.name }}
+        path: ${{ inputs.path }}
+        list-suites: ${{ inputs.list-suites }}
+        list-tests: ${{ inputs.list-tests }}
+        fail-on-error: ${{ inputs.fail-on-error }}
+        only-summary: ${{ inputs.only-summary }}
```

Comment:
these inputs are only used in this report, I am not sure if it is better to use an `options` input and define the extra parameter in YAML to JSON in that field.

```yaml
jobs:
  process:
    runs-on: ubuntu-latest
    steps:
...
      - uses: elastic/apm-pipeline-library/.github/actions/test-report@main
        name: foo
        path: /bar
        options: |
          op01: val01
          op02: var02
```

Reply:
with JSON the transformation is direct https://docs.github.com/en/actions/learn-github-actions/expressions#tojson with YAML we have to implement a simple action to read a YAML input-key

Reply:
or transform YAML to JSON 
```bash
OPTIONS=${{ input.options }}
JSON=$(echo "${OPTIONS}" | yq -c .) 
echo "json=${JSON}" >> $GITHUB_OUTPUT 
```